core/2d/CCFastTMXLayer.cpp

```diff
@@ -356,7 +356,16 @@ void FastTMXLayer::setupTiles()
                 }
 
                 int pos = static_cast<int>(newX + _layerSize.width * y);
-                int gid = _tiles[pos];
+                uint32_t gid = _tiles[pos];
+                uint32_t flags = 0;
+
+                flags |= gid & kTMXTileHorizontalFlag;
+                flags |= gid & kTMXTileVerticalFlag;
+                flags |= gid & kTMXTileDiagonalFlag;
+
```

Comment:
It may be better to just do this:
`flags |= gid & kTMXFlipedAll;`

since `kTMXFlipedAll          = (kTMXTileHorizontalFlag | kTMXTileVerticalFlag | kTMXTileDiagonalFlag)`

Reply:
Good catch!